Gadgets/MatchingGadgets.h

```diff
@@ -601,47 +370,44 @@ class OrderMatchingGadget : public GadgetT
     CheckValidGadget checkValidA;
     CheckValidGadget checkValidB;
     AndGadget valid;
+    RequireEqualGadget requireValid;
 
     OrderMatchingGadget(
         ProtoboardT& pb,
         const Constants& constants,
         const VariableT& timestamp,
         const OrderGadget& orderA,
         const OrderGadget& orderB,
+        const VariableT& _fillS_A,
+        const VariableT& _fillS_B,
         const std::string& prefix
     ) :
         GadgetT(pb, prefix),
+        fillS_A(_fillS_A),
+        fillS_B(_fillS_B),
 
-        // Get the max amount the orders can be filled
-        maxFillAmountA(pb, constants, orderA, FMT(prefix, ".maxFillAmountA")),
-        maxFillAmountB(pb, constants, orderB, FMT(prefix, ".maxFillAmountB")),
-
-        // Match the orders
-        matchingGadget(pb, constants,
```

Comment:
The only concern of removing matchingGadget is sometimes FillS_A/FillS_B < makerB/makerS, which looks like the trade is done without meet the maker's price requirement, although this may be not a problem since the error is really small as it is limited to Float24Encoding.

Reply:
Even with `matchingGadget` there could be some rounding errors, but yeah they will be really small.

And with this PR we don't require the trade to happen at the maker's order. It's still possible, but not required.